src/Inciter/AMR/Refinement_State.h

```diff
@@ -0,0 +1,167 @@
+//
+// Created by Robert Francis Bird on 10/27/16.
+//
+
+#ifndef TETAMR_AMR_DATA_H
+#define TETAMR_AMR_DATA_H
+
+#include <vector>
+#include <cstddef>
+
+#include "types.h"
+
+namespace AMR {
+    // Types
+    using real_t = tk::real;
+
+    const size_t DEFUALT_CHILD_NUMBER = 0; // Used to mark "doesn't have children"
+    const size_t DEFAULT_NUM_CHILDREN = 0; // Default to no children
+
+    // TODO: Populate this enum with the available refinement cases
+    enum Refinement_Case { initial_grid, one_to_two, one_to_four, one_to_eight,
+        two_to_eight, four_to_eight, none };
+
+    enum Edge_Lock_Case {unlocked, locked, intermediate, temporary};
+
+    struct Edge_Refinement {
+        size_t A;
+        size_t B;
+        real_t refinement_criteria;
+        bool needs_refining; // TODO: This could possibly be deduced implicitly
+        bool needs_derefining; // TODO: Marge this with needs_refining
+        bool is_dead;
+        Edge_Lock_Case lockCase; // TODO: Refactor this to match _ style?
+
+        // Explicit Empty Constructor
+        Edge_Refinement() {
+            A = 0;
+            B = 0;
+            refinement_criteria = 0.0;
+            needs_refining = false;
+            needs_derefining = false;
+            is_dead = false;
+            lockCase = Edge_Lock_Case::unlocked;
+        }
+
+        // Explicit copy Constructor
+        Edge_Refinement(const Edge_Refinement& a) :
+            A(a.A),
+            B(a.B),
+            refinement_criteria(a.refinement_criteria),
+            needs_refining(a.needs_refining),
+            needs_derefining(a.needs_derefining),
+            is_dead(a.is_dead),
+            lockCase(a.lockCase)
+        {
+            // Empty
+        }
+
+        // This abstraction is hardly any better than using an explicit initialisation
+        // list but it makes it easier if we decide to add/remove a parameter
+        Edge_Refinement(
+                size_t A_in,
+                size_t B_in,
+                real_t refinement_criteria_in,
+                bool needs_refining_in,
+                bool needs_derefining_in,
+                bool is_dead_in,
+                Edge_Lock_Case lockCase_in
+                ) :
+            A(A_in),
+            B(B_in),
+            refinement_criteria(refinement_criteria_in),
+            needs_refining(needs_refining_in),
+            needs_derefining(needs_derefining_in),
+            is_dead(is_dead_in),
+            lockCase(lockCase_in)
+        {
+            // Empty, all implicit.
+            // Could add logic here to reconcile needs_refining and needs_derefining
+        }
+    };
+
+    //stop it being copied around?
+
+    class Refinement_State {
```

Comment:
The same as [above](https://github.com/quinoacomputing/quinoa/pull/156/files#r131698931) should work here too: I would do `struct` and only put in the state and perhaps a constructor with an initializer list with the defaults. Then let the compiler auto-generate everything else as needed.